app/javascript/components/users/authentication/forms/SignupForm.jsx

```diff
@@ -1,62 +1,71 @@
-import React, { useState, useRef } from 'react';
+import React, { useRef, useMemo, useCallback } from 'react';
 import { Button, Container, Stack } from 'react-bootstrap';
-import { useForm } from 'react-hook-form';
 import HCaptcha from '@hcaptcha/react-hcaptcha';
 import { toast } from 'react-hot-toast';
 import { useTranslation } from 'react-i18next';
 import FormControl from '../../../shared_components/forms/FormControl';
 import Form from '../../../shared_components/forms/Form';
-import { signupFormConfig, signupFormFields } from '../../../../helpers/forms/SignupFormHelpers';
 import Spinner from '../../../shared_components/utilities/Spinner';
 import useCreateUser from '../../../../hooks/mutations/users/useCreateUser';
 import useEnv from '../../../../hooks/queries/env/useEnv';
+import useSignUpForm from '../../../../hooks/forms/authentication/useSignUpForm';
 
 export default function SignupForm() {
+  const captchaRef = useRef(null);
   const { t } = useTranslation();
-  const methods = useForm(signupFormConfig);
-  const [token, setToken] = useState('');
-  const { onSubmit: createUser } = useCreateUser(token);
-  const { isSubmitting } = methods.formState;
-  const fields = signupFormFields;
+  const { fields, methods } = useSignUpForm();
```

Comment:
Instead of having to deal with multiple objects and to do multiple steps to set a form,
The idea is now similar to how react hook form was designed.
Any form low logic on the App is nothing but a hook,
The hook will configure the react hook form as intended,
Plus extend that with the fields API that the Form controls API will use seamlessly,